apps/sms/test/unit/navigation_test.js

```diff
@@ -521,6 +521,66 @@ suite('navigation >', function() {
     });
   });
 
+  suite('ensureCurrentPanel >', function() {
+    setup(function(done) {
+      this.sinon.stub(Navigation, 'slide').returns(Promise.resolve());
+      this.sinon.spy(Navigation, 'toPanel');
+
+      // Set custom hash to have non-initialized current panel
+      window.location.hash = '#x-unknown-panel';
+
+      Navigation.init().catch((e) => {
+        Navigation.toPanel.reset();
+
+        // It's expected error for the unknown panel
+        if (e.message === 'Panel x-unknown-panel is unknown.') {
+          return;
+        }
+
+        return Promise.reject(e);
```

Comment:
Ha-ha, you're right! As always I want JS to be like C#, but here it's more like Java that preserves original stack trace when rethrown. My bad :)